design/unnumbered-bgp.md

```diff
@@ -0,0 +1,498 @@
+in IPv4 setup (as long as the IPv6 stack/LLAs are not disabled).
+
+```
+> docker exec -it unnumbered-p2p-peer vtysh -c 'show bgp ipv4'
+BGP table version is 3, local router ID is 11.11.11.254, vrf id 0
+Default local pref 100, local AS 65004
+Status codes:  s suppressed, d damped, h history, * valid, > best, = multipath,
+               i internal, r RIB-failure, S Stale, R Removed
+Nexthop codes: @NNN nexthop's vrf id, < announce-nh-self
+Origin codes:  i - IGP, e - EGP, ? - incomplete
+RPKI validation codes: V valid, I invalid, N Not found
+
+    Network          Next Hop            Metric LocPrf Weight Path
+ *= 5.5.5.5/32       eth02                    0             0 65000 i
+ *=                  eth01                    0             0 65000 i
+ *>                  eth00                    0             0 65000 i
+ *> 200.100.100.0/24 0.0.0.0(unnumbered-p2p-peer)
+                                             0         32768 ?
+
+Displayed  2 routes and 4 total paths
+> docker exec -it unnumbered-p2p-peer ip route
+5.5.5.5 nhid 21 proto bgp metric 20
+        nexthop via inet6 fe80::dcad:beff:feff:1160 dev eth00 weight 1
+        nexthop via inet6 fe80::dcad:beff:feff:1161 dev eth01 weight 1
+        nexthop via inet6 fe80::dcad:beff:feff:1162 dev eth02 weight 1
+```
+
+### The API Extension
+
+We have three option:
+
+#### Option 1. Re-Use Address field
+
+We can re-use `Address string` to either be IP or interface, if not valid IP we
+indicate Unnumbered BGP peering. This looks the minimal API, but it will not
+have clean error path. E.g. if the user input is not valid IP, the
+implementation will shadow all the existing errors and will fail in the
+runtime.
+
+#### Option 2. Add a new Interface field
+
+We can introduce a new field `Interface string` which if defined then Unnumber
+BGP takes place. In that case, the address must NOT have value. For
+completeness, we could use field might be used to support `LLA%zone` usecase.
+So at the end, the API doc will look like
+
+```
+// 1. if address and no interface then as normal
+// 2. if non valid address and no interface then fail
+// 3. if address is LLA and interface not empty, then LLA support
+// == extra FRR directive `neighbor fe80::a876:4dff:fe77:408 interface eth0`
+// 4. If address is LLA and empty interface, then fail
+// 5. If address is empty and interface has value then the BGP unnumber peering config will take place
+// ==  modify FRR directive `neighbor PEER interface remote-as <ASN>`
+// && replace any address directive to use interface e.g.  `neighbor net0 timers 180 540`
+// && enable RAs on this interface
+// 6. If address has value (non-LLA) and interface non empty, then fail
+```
+That option might be not ideal (subjective opinion) because documentation looks complicated,
+and user might have to do trial-and-error approach. Plus user needs to understand corner cases.
```

Comment:
the user will need to understand the usecases as it does for frr, is there extra complexity beyond that?

Reply:
imo yes :)

Reply:
so I think it should be more detailed, because it is not really clear atm